module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -106,137 +108,167 @@ namespace module::vision {
+                        radius = uBCw.dot(uECw) < radius ? uBCw.dot(uECw) : radius;
                     }
 
-                    // Ball cam space info
-                    b.cone.axis   = horizon.Hcw.topLeftCorner<3, 3>().cast<float>() * axis;
-                    b.cone.radius = radius;
+                    // The vectors are in world space, multiply by Rcw to get the central axis in camera space
+                    b.uBCc   = Hcw.rotation() * uBCw;
+                    b.radius = radius;  // arccos(radius) is the angle between the central axis and edge of the ball
+
+                    // CALCULATE DISTANCE TO BALL WITH TWO METHODS
+                    // 1. Angular-based distance
+                    // 2. Projection-based distance
 
+                    // Angular-based distance
                     // https://en.wikipedia.org/wiki/Angular_diameter
+                    // From the link, the formula with arcsin is used since a ball is a spherical object
+                    // The variables are:
+                    //      delta: 2 * arccos(radius)
+                    //      d_act: 2 * field.ball_radius
+                    //      D: distance
+                    // Rearranging the equation gives
+                    //      distance = (field.ball_radius) / (sin(arccos(radius)))
+                    // Using sin(arccos(x)) = sqrt(1 - x^2)
+                    //      distance = field.ball_radius / sqrt(1 - radius^2)
                     float distance = field.ball_radius / std::sqrt(1.0f - radius * radius);
+                    // Convert uBCc into a position vector (rBCc) and then convert it into Spherical Reciprocal
+                    // Coordinates (1/distance, phi, theta)
+                    b.srBCc              = cartesianToReciprocalSpherical(b.uBCc * distance);
+                    b.screen_angular_BCw = cartesianToSpherical(uBCw).tail<2>();  // no depth (x-coordinate)
+
+                    // Projection-based distance
+                    // Given a flat X-Y plane that intersects through the middle of the ball, find the point where
+                    // the central axis vector (uBCw) intersects with this place. Get the distance from this point.
+                    // https://en.wikipedia.org/wiki/Line%E2%80%93plane_intersection#Algebraic_form
+                    // Plane normal = (0, 0, 1)
+                    // Point on plane = (0, 0, field.ball_radius) // SHOULD THIS BE (0,0, BALL RADIUS + rWCw)
+                    // Line direction = uBCw
+                    // Point on line = camera = Hwc.translation.z() = rCWw.z() // SHOULD THIS BE (0,0,0)
+                    // Since the plane normal zeros out x and y, only consider z
+                    const float d = (Hcw.inverse().translation().z() - field.ball_radius) / std::abs(uBCw.z());
+                    const Eigen::Vector3f srBCc     = uBCw * d;  // used later, so save it in a variable
```

Comment:
@TrentHouliston 
I've updated the variable names to hopefully match conventions.
It seems the angular based distance is determined using world space but centered on the camera (like using uXCw type vectors)
But projection based seems to be going straight from world. My thoughts on what projection based should be is that 
- the line is ofc uBCw (central axis vector), 
- the point on the line is (0,0,0) since the line is on the camera and we're measuring from the camera (in world space)
- the point on the plane is like (ball_radius + rWCw) because from the camera, go down to the world and then up to the ball radius

Given this, the equation becomes $d = \frac{p_0.z()}{l.z()} = \frac{rBCw.z()}{\text{uBCw.z()}}$
What is in the code is pretty much the same, since it's $\frac{rCBw.z()}{abs(uBCw.z()}$
the way in the code makes the top thing positive and abs the bottom, but since those negatives will cancel anyway it might make more sense to go with the straight derivation.

Reply:
Pretty sure `srBCc` should just be `rBCc` here since it is in cartesian coordinates not spherical coordinates.

Although your method results in a very similar equation I think the logic on getting there is a little more convoluted. The way that is currently implemented, the plane is the field (with a slightly higher z-value so it passes through the ball) and the camera is above is the field. I feel like this is easier to visualise.

Reply:
I think this comes down to personal preference. The original stuff seemed more convoluted to me.
Maybe I'm not understanding the original one well enough, but it seems like it doesn't quite following the spaces correctly and then has to do an abs() to fix up the signs.
The way I thought about it was going from the formula and making sure all the spaces were correct throughout. 
In the end it is the same thing though. I'm just not sure how to nicely comment the original stuff to show how you get it mathematically.

Reply:
Everything is in world space, so thats not an issue. Its just a question of where your origin is, is it at camera height or ball height? And it really doesn't matter which one you choose, so it becomes a question of which version is easier to understand. 

You are projecting a ray from the camera to the "ground" plane so, to me, having the plane at "ground" level makes more sense. Then, the camera being above the plane follows from that. The alternative is your suggestion, where the ground is below the camera, but as I said either option is equally correct.